### PR TITLE
Checkbox

### PR DESCRIPTION
**Pull request by aurelienbottazini**
https://www.pivotaltracker.com/story/show/143953795

image

Merge
https://github.com/doximity/vue-genome/pull/58
and 
https://github.com/doximity/vue-genome/pull/56
Before this one

### CODE REVIEW

File: src/components/buttons/Checkbox.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      checked: {
+        type: Boolean,
+        default: false,
+      },
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    methods: {
```
Comment (Austio) [inline src/components/buttons/Checkbox.vue]: Have you considered how this be used with v-model?

Convention we have been using: change/input emit the value, all others emit the event.

Comment (aurelienbottazini) [inline src/components/buttons/Checkbox.vue]: what do you think about refactoring using a mixin to handle this?

Comment (Austio) [inline src/components/buttons/Checkbox.vue]: Is good idea, there will be cases where we will input at different times and in different contexts (for instance in autocomplete/select we input on a selected item whereas for input-box it is when a user types in the box).  I think just do manually now and then extract out later if we find it cumbersome.

Comment (aurelienbottazini) [inline src/components/buttons/Checkbox.vue]: 👍

File: src/components/buttons/Checkbox.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      handleClick(event) {
+        this.$emit('click', event);
+      },
+      handleChange(event) {
+        this.$emit('input', event.target.checked);
+        this.$emit('change', event.target.checked);
+      },
```
Comment (aurelienbottazini) [inline src/components/buttons/Checkbox.vue]: on checkbox, @input does not work, we have to pass through @change

Comment (Austio) [inline src/components/buttons/Checkbox.vue]: :+1: i think both of these make sense in context of checkbox

**Pull request merged**